packets.js

```diff
@@ -127,21 +127,57 @@ runknown.encodingLength = function (data) {
 
 var rtxt = {}
 
-rtxt.encode = function (s, buf, offset) {
-  str.encode(s, buf, offset + 2)
-  buf.writeUInt16BE(str.encode.bytes, offset)
-  rtxt.encode.bytes = str.encode.bytes + 2
+rtxt.encode = function (data, buf, offset) {
+  var oldOffset = offset
+  offset += 2
+
+  if (!data) data = {}
+  var keys = Object.keys(data)
+
+  if (keys.length === 0) {
+    str.encode('', buf, offset)
+    offset += str.encode.bytes
+  }
+
+  keys.forEach(function (key) {
+    var s = key + '=' + data[key]
```

Comment:
I didn't look at the code when I wrote the above comment - wonderful how your brain can forget what you've already coded :wink: The code in fact does follow the RFC. So all that's left is to allow the encoder to encode `{key: true}` to just `key` - I'll get on it.